azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -152,13 +152,6 @@ def _fetch_manifest(self, manifest_type, name, uris):
         except Exception as e:
             raise ProtocolError("Failed to retrieve {0} manifest. Error: {1}".format(manifest_type, ustr(e)))
 
-    def download_extension(self, uris, destination, on_downloaded=lambda: True):
```

Comment:
Initially I added this method here for consistency with fetch_extension_manifest above, but it really doesn't quite fit here so I'm removing it.